SSA/Core/MLIRSyntax/Parser.lean

```diff
@@ -0,0 +1,132 @@
+import SSA.Core.MLIRSyntax.AST
+import SSA.Core.MLIRSyntax.EDSL
+import Lean
+
+open Lean
+
+variable {ParseOutput : Type} [ToString ParseOutput]
+
+abbrev ParseError := String
+abbrev ParseFun : Type := Lean.Environment → String → EIO ParseError ParseOutput
+
+-- see https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/Calling.20unification.20from.20meta.20to.20remove.20metavariables
```

Comment:
If we link to a zulip thread, can we also commit a 1-2 sentence summary of the thread to the repo, such that we know why it is linked.